src/test/java/GamePlayTest.java

```diff
@@ -52,70 +45,187 @@ public static void tearDown() throws Exception {
-        pauseOverlayPage.getResumeButton();
         pauseOverlayPage.pressMainMenu();
         assertTrue(mainMenuPage.isDisplayed());
     }
 
     @Test
     public void testAvoidingObstacles() throws Exception {
         mainMenuPage.pressRun();
-        gamePlayPage.getCharacter();
-        gamePlayPage.getPauseButton();
-        gamePlayPage.avoidObstacles(0);
-        assertTrue(gamePlayPage.getCurrentLife()>=0);
+        gamePlayPage.avoidObstacles(5);
+        System.out.println("Current life after avoiding obstacles: " + gamePlayPage.getCurrentLife());
+        assertTrue(gamePlayPage.getCurrentLife() > 0);
     }
 
     @Test
     public void testPlayerDiesWhenObstacleNotAvoided() throws Exception {
-
         mainMenuPage.pressRun();
-        gamePlayPage.getCharacter();
-        gamePlayPage.getPauseButton();
-
         float timeout = 20;
-        while(timeout>0){
+        while (timeout > 0) {
             try {
-                getAntoherChancePage.getGameOver();
-                getAntoherChancePage.getAvailableCurrency();
-                getAntoherChancePage.getPremiumButton();
-                getAntoherChancePage.isDisplayed();
+                getAnotherChancePage.isDisplayed();
                 break;
-            }catch(Exception e){
+            } catch (Exception e) {
                 timeout -= 1;
             }
         }
+    }
+
+    @Test
+    public void testDistanceRun() throws Exception {
+        mainMenuPage.pressRun();
+
+        AltObject characterStart = gamePlayPage.getCharacter();
+
+        List<Float> zValues = new ArrayList<>();
+        long startTime = System.currentTimeMillis();
+        long duration = 25000;
+        long interval = 1000;
+        long nextTimestamp = startTime + interval;
+
+        while (System.currentTimeMillis() - startTime < duration) {
+            try {
+                getAnotherChancePage.isDisplayed();
+                break;
+            } catch (Exception e) {
+                if (System.currentTimeMillis() >= nextTimestamp) {
+                    AltObject character = gamePlayPage.getCharacter();
+                    zValues.add(character.worldZ);
+                    nextTimestamp += interval;
+                }
+            }
+        }
+
+        AltObject characterFinal = gamePlayPage.getCharacter();
+
+        int distanceRun = gamePlayPage.getDistanceRun();
+
+        float resetValue1 = 100.0f;
+        float resetValue2 = 200.0f;
+
+        int resetCount = 0;
+        for (int i = 1; i < zValues.size(); i++) {
+            if (zValues.get(i) < zValues.get(i - 1)) {
+                resetCount++;
+            }
+        }
+
+        float zDistance = 0;
+        if (resetCount == 1) {
+            zDistance = resetValue1 + characterFinal.worldZ - 2;
+        } else if (resetCount == 2) {
+            zDistance = resetValue2 + characterFinal.worldZ - 2;
+        } else {
+
+            zDistance = characterFinal.worldZ - characterStart.worldZ;
+        }
```

Comment:
this should be: zDistance= resetCount* 100 + characterFinal.worldZ-1;

This way it is correct also if the resetCount is higher than 2

Reply:
Refactored in e9d7437